openmdao/core/problem.py

```diff
@@ -388,13 +388,15 @@ def _get_ubc_vars(self, connections):
                      enumerate(self.root.subsystems(recurse=True))}
 
         ubcs = []
+        tgts = []
```

Comment:
I would just make this a set to begin with instead of creating a list that you later convert into a set.  There's no point in having the get_ubc_vars function return a list of target systems containing duplicates.